docker-alfresco/test/docker-alfresco-all-amps-test/scripts/checkLibraryDuplicates.sh

```diff
@@ -2,24 +2,47 @@
 
 # Script used to check for duplicated libraries after applying all amps
 
-Libdir=$1
-lib_list=""
-echo "Scanning libraries from $Libdir"
+# List of library names (separated by space) for which the version is not easy to identify with simple pattern matching,
+# usually libraries with versions that don't follow naming/versioning standards.
+# e.g. geronimo-jms_2.0_spec-1.0-alpha-2.jar
+#      geronimo-jms_1.1_spec-1.1.1.jar
+WHITELIST="geronimo-jms"
 
-for lib1 in $(ls "$Libdir"); do
-    if [[ "$lib1" =~ [0-9]+.[0-9] ]]; then
-        noversion="${lib1%%"$BASH_REMATCH"*}"
-        for lib2 in $(ls -I "$lib1" "$Libdir"); do
-            if [[ "$lib2" = "$noversion"[0-9].* ]]; then
-                lib_list="$lib_list $lib1"
-            fi
-        done
-    fi
+lib_dir=$1
+multiple_version_lib_list=""
+
+echo "Scanning libraries from $lib_dir"
+if [[ -n "$WHITELIST" ]]; then
+    echo "Skipping libraries which match the whitelist, check for false positives:"
+fi
+
+for current_lib in $(ls "$lib_dir"); do
+   if [[ "$current_lib" =~ [0-9]+.[0-9] ]]; then
+       noversion_lib="${current_lib%%"$BASH_REMATCH"*}"
+
+       skip=false
+       for exclude_lib in $WHITELIST; do
+           if [[ "$noversion_lib" =~ "$exclude_lib" ]]; then
+               skip=true
+               break
+           fi
+       done
+       if [[ "$skip" = true ]]; then
+           echo "Skip $current_lib"
+           continue
+       fi
+       
+       for other_lib in $(gls --ignore="$current_lib" "$lib_dir"); do
```

Comment:
The proper command for linux is `ls` not `gls`, this was fixed on the 6.0.N branch, after the merge was done.